study-frontend/src/Components/HomePageInformation.tsx

```diff
@@ -0,0 +1,150 @@
+import React, {useEffect, useState} from 'react'
+import { MdChevronRight } from "react-icons/md";
+import { MdKeyboardArrowLeft } from "react-icons/md";
+import exampleImage1 from '../assets/continious-learning.png';
+import onlineEdu from '../assets/onlineEdu.png'
+import TeacherCard from './TeacherCard';
+import { FaRegCircle } from "react-icons/fa";
+
+
+interface Tag{
+    id : number,
+    name : string,
+    color : string 
+  }
+
+interface User {
+    id : number,
+    username: string,
+    email : string
+    tags : Array<Tag>
+    description : string, 
+    rating : number
+}
+  
+type Props = {
+    user: User
+}
+
+const HomePageInformation = ({user}: Props) => {
+
+    const [sliderIndex, setIndex] = useState(1);
+
+    const [viewportWidth, setViewportWidth] = useState(window.innerWidth);
+  
+    React.useEffect(() => {
+  
+      window.addEventListener("resize", () => setViewportWidth(window.innerWidth));
```

Comment:
Its better to move this logic into a higher component lets say the app state or a Theme state.
Because you will need this everywhere and mostlikely you will have to rerender the app.